Foundation/Operation.swift

```diff
@@ -398,15 +391,16 @@ open class OperationQueue: NSObject {
         addOperations([op], waitUntilFinished: false)
     }
     
-    internal func _runOperation() {
-        if let op = _dequeueOperation() {
-            if !op.isCancelled {
-                op._waitUntilReady()
-                if !op.isCancelled {
-                    op.start()
-                }
-            }
+    internal func _dequeueReadyOperation() -> Operation? {
+        guard let op = _dequeueOperation() else {
+            return nil
+        }
+
+        if !op.isCancelled {
+            op._waitUntilReady()
```

Comment:
this isnt exactly how it works, the dequeue should be re-triggered upon finished states. The current implementation is a bit simplified since we dont have KVO.